lib/pbench/server/cache_manager.py

```diff
@@ -20,93 +20,80 @@
 class CacheManagerError(Exception):
     """Base class for exceptions raised from this module."""
 
-    def __str__(self) -> str:
-        return "Generic cache manager exception"
+    pass
 
 
 class BadDirpath(CacheManagerError):
     """A bad directory path was given."""
 
     def __init__(self, error_msg: str):
+        super().__init__(error_msg)
         self.error_msg = error_msg
 
-    def __str__(self) -> str:
-        return self.error_msg
-
 
 class BadFilename(CacheManagerError):
     """A bad tarball path was given."""
 
     def __init__(self, path: PathLike):
+        super().__init__(f"The file path {path} is not a tarball")
         self.path = str(path)
 
-    def __str__(self) -> str:
-        return f"The file path {self.path!r} is not a tarball"
```

Comment:
The problem was that `self.path` is a `Path`, not a `str`, so `!r` here wasn't really doing what was originally intended. I could have used `'{self.path}'` but chose not to bother. I was trying to keep most of the message strings the same, although in the process I cleaned up some inconsistencies like this.

`TarballNotFound` takes a `str`, so `{tarball!r}` was used correctly.

Reply:
Right...so, rather than dropping the `!r` I was wondering whether you might be inclined to add a `str()`:
```python
        super().__init__(f"The file path {str(path)!r} is not a tarball")
```
which would produce the quotes.